mqtt/mqtt-bridge/src/client.rs

```diff
@@ -458,16 +458,147 @@ impl ShutdownHandle {
     }
 }
 
+pub struct MqttClientConfig {
+    addr: String,
+    keep_alive: Duration,
+    clean_session: bool,
+    credentials: Credentials,
+    max_in_flight: usize,
+}
+
+impl MqttClientConfig {
+    pub fn new(
+        addr: impl Into<String>,
+        keep_alive: Duration,
+        clean_session: bool,
+        credentials: Credentials,
+        max_in_flight: usize,
+    ) -> Self {
+        Self {
+            addr: addr.into(),
+            keep_alive,
+            clean_session,
+            credentials,
+            max_in_flight,
+        }
+    }
+}
+
+/// Trait used to facilitate mock publish handle types
+#[async_trait]
+pub trait InnerPublishHandle {
+    async fn publish(&mut self, publication: Publication) -> Result<(), PublishError>;
+}
+
 /// A client publish handle.
+#[derive(Debug, Clone)]
 pub struct PublishHandle(mqtt3::PublishHandle);
 
-#[automock]
-impl PublishHandle {
-    pub async fn publish(&mut self, publication: Publication) -> Result<(), PublishError> {
+#[async_trait]
+impl InnerPublishHandle for PublishHandle {
+    async fn publish(&mut self, publication: Publication) -> Result<(), PublishError> {
         self.0.publish(publication).await
     }
 }
 
+/// A wrapper around mockall publish handle necessary for cloning
+#[derive(Clone)]
+pub struct MockallPublishHandleWrapper {
+    inner: Arc<Mutex<MockPublishHandle>>,
+}
+
+impl MockallPublishHandleWrapper {
+    pub fn new() -> Self {
+        let inner = Arc::new(Mutex::new(MockPublishHandle::new()));
+        Self { inner }
+    }
+
+    pub fn inner(&self) -> Arc<Mutex<MockPublishHandle>> {
+        self.inner.clone()
+    }
+}
+
+#[async_trait]
+impl InnerPublishHandle for MockallPublishHandleWrapper {
+    async fn publish(&mut self, publication: Publication) -> Result<(), PublishError> {
+        self.inner.lock().await.publish(publication).await
+    }
+}
+
+mock! {
+    pub PublishHandle {}
+
+    #[async_trait]
+    pub trait InnerPublishHandle {
+        async fn publish(&mut self, publication: Publication) -> Result<(), PublishError>;
+    }
+}
```

Comment:
I tried using automock, but there is weird behavior with the test where it thinks the automock is for InFlightPublishHandle. Having a more explicit mock fixes this error.